qiskit/algorithms/evolvers/classical_methods/scipy_imaginary_evolver.py

```diff
@@ -0,0 +1,192 @@
+            evolution_problem=evolution_problem
+        )
+
+        # Create empty arrays to store the time evolution of the aux operators.
+        operator_number = (
+            0 if evolution_problem.aux_operators is None else len(evolution_problem.aux_operators)
+        )
+        ops_ev_mean = np.empty(shape=(operator_number, self.timesteps + 1), dtype=float)
+
+        ops_ev_std = np.empty(shape=(operator_number, self.timesteps + 1), dtype=float)
+
+        for ts in range(self.timesteps):
+            (ops_ev_mean[:, ts], ops_ev_std[:, ts],) = self._evaluate_aux_ops(
+                aux_ops,
+                aux_ops_2,
+                state,
+            )
+            state = expm_multiply(A=-hamiltonian * timestep, B=state)
+            if np.nan in state:
+                raise ValueError(
+                    "An overflow has probably occured. Try increasing the amount of timesteps."
+                )
+            state = state / np.linalg.norm(state)
+
+        (ops_ev_mean[:, self.timesteps], ops_ev_std[:, self.timesteps],) = self._evaluate_aux_ops(
+            aux_ops,
+            aux_ops_2,
+            state,
+        )
+
+        aux_ops_history = self._create_observable_output(
+            ops_ev_mean,
+            ops_ev_std,
+            evolution_problem.aux_operators,
+        )
+        aux_ops = self._create_observable_output(
+            ops_ev_mean[:, self.timesteps],
+            ops_ev_std[:, self.timesteps],
+            evolution_problem.aux_operators,
+        )
+
+        return EvolutionResult(
+            evolved_state=StateFn(state), aux_ops_evaluated=aux_ops, observables=aux_ops_history
+        )
+
+    def _create_observable_output(
+        self,
+        ops_ev_mean: np.ndarray,
+        ops_ev_std: np.ndarray,
+        aux_ops: ListOrDict,
+    ) -> ListOrDict[Union[Tuple[np.ndarray, np.ndarray], Tuple[complex, complex]]]:
+        """Creates the right output format for the evaluated auxiliary operators."""
+        operator_number = 0 if aux_ops is None else len(aux_ops)
+        observable_evolution = [(ops_ev_mean[i], ops_ev_std[i]) for i in range(operator_number)]
+
+        if isinstance(aux_ops, dict):
+            observable_evolution = dict(zip(aux_ops.keys(), observable_evolution))
+
+        return observable_evolution
+
+    def _evaluate_aux_ops(
+        self,
+        aux_ops: List[sp.csr_matrix],
+        aux_ops_2: List[sp.csr_matrix],
+        state: np.ndarray,
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        """Evaluates the aux operators if they are provided and stores their value.
+
+        Returns:
+            Tuple of the mean and standard deviation of the aux operators for a given state.
+        """
+        op_mean = np.array([np.real(state.conjugate().dot(op.dot(state))) for op in aux_ops])
+        op_std = np.sqrt(
+            np.array([np.real(state.conjugate().dot(op2.dot(state))) for op2 in aux_ops_2])
+            - op_mean**2
+        )
```

Comment:
I thought the stddev in the aux operator results is the stddev in the measurements, not the stddev of the expectation value. For shots this _should_ be the same, but for statevector simulations the measurement stddev is always 0. What does the VQE do here for statevector simulations, or the NumPyEigensolver?

Reply:
You are right, the standard deviation should be just 0.0.